File: Part2.py
def main():
    input_line = input().strip()
    parts = input_line.split(",")

    cleaned_parts = []
    for item in parts:
        item = item.strip()
        cleaned_parts.append(item)

    unique_set = set()

    for item in cleaned_parts:
        unique_set.add(item)

    count_map = {}

    for item in cleaned_parts:
        if item in count_map:
            new_value = count_map[item] + 1
            count_map[item] = new_value
        else:
            count_map[item] = 1

    print("List:", cleaned_parts)
    print("Set:", unique_set)
    print("Dictionary:", count_map)


if __name__ == "__main__":
    main()

File: Part2.java
import java.util.*;

public class Part2 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String input = scanner.nextLine();

        String[] parts = input.split(",");

        ArrayList<String> list = new ArrayList<>();
        for (String item : parts) {
            list.add(item.strip());  
        }

        HashSet<String> set = new HashSet<>();
        for (String item : list) {
            set.add(item);
        }
        HashMap<String, Integer> map = new HashMap<>();
        for (String item : list) {
            if (map.containsKey(item)) {
                map.put(item, map.get(item) + 1);
            } else {
                map.put(item, 1);
            }
        }

        System.out.println("ArrayList: " + list);
        System.out.println("HashSet: " + set);
        System.out.println("HashMap: " + map);

        scanner.close();
    }
}
